app/RouteHandlers/PayPalNotificationHandler.php

```diff
@@ -63,10 +63,14 @@ private function getValueFromCustomVars( string $customVars, string $key ): stri
 	}
 
 	private function requestIsForPaymentCompletion( ParameterBag $post ): bool {
-		return $this->isSuccessfulPaymentNotification( $post ) || (
-				$this->isForRecurringPayment( $post ) &&
-				!$this->isRecurringPaymentCompletion( $post )
-			);
+		if ( !$this->isSuccessfulPaymentNotification( $post ) ) {
+			return false;
+		}
+		if ( $this->isForRecurringPayment( $post ) && !$this->isRecurringPaymentCompletion( $post ) ) {
+			return false;
+		}
+
+		return true;
```

Comment:
\o/ much more readable. The amount of negations (including in the calling predicate) is still rather mindfuck tho